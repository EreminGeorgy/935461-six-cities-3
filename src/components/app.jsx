import React from "react";
import Main from "./main.jsx";


const App = (props) => {
  // eslint-disable-next-line react/prop-types
  const {proposalsNumber} = props;

  return (
    <Main proposalsNumber={proposalsNumber} />
  );
};


export default App;
