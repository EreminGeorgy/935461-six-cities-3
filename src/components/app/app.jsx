import React, {useState} from "react";
import {BrowserRouter, Switch, Route, Redirect} from 'react-router-dom';
import Main from "../main/main.jsx";
import {Property} from "../property/property.jsx";
import SignIn from "../sign-in/sign-in.jsx";
import {connect} from "react-redux";
import {Operation} from "../../reducer/user/user.js";
import PropTypes from "prop-types";
import {getAuthorizationStatus} from "../../reducer/user/selectors.js";
import {AuthorizationStatus} from "../../reducer/user/user";

export const App = (props) => {
  const {login, authorizationStatus} = props;

  const [activeOffer, setActiveOffer] = useState(null);

  return (
    <BrowserRouter>
      <Switch>
        <Route exact path="/">
          <Main
            handleTitleClick={setActiveOffer}
          />
        </Route>
        <Route path="/dev-property">
          <Property
            offer={activeOffer}
          />
        </Route>
        <Route exact path="/dev-auth">
          {(authorizationStatus === AuthorizationStatus.AUTH) ?
            <Redirect to={`/`} /> :
            <SignIn
              onSubmit={login}
            />
          }
        </Route>
      </Switch>
    </BrowserRouter>
  );
};

const mapStateToProps = (state) => ({
  authorizationStatus: getAuthorizationStatus(state),
});

const mapDispatchToProps = (dispatch) => ({
  login(authData) {
    dispatch(Operation.checkAuth(authData));
  },
});

export default connect(mapStateToProps, mapDispatchToProps)(App);

App.propTypes = {
  login: PropTypes.func,
  authorizationStatus: PropTypes.string,
};
