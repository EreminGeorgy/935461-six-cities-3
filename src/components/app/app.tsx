import * as React from "react";
import {BrowserRouter, Switch, Route, Redirect} from 'react-router-dom';
import Main from "../main/main";
import Property from "../property/property";
import SignIn from "../sign-in/sign-in";
import Favorites from "../favorites/favorites";
import PrivateRoute from "../private-route/private-route";
import {connect} from "react-redux";
import {Operation} from "../../reducer/user/user";
import {Offer} from "../../types";
import {getAuthorizationStatus} from "../../reducer/user/selectors";
import {AuthorizationStatus} from "../../reducer/user/user";
import {AppRoute} from "../../utils/const";

interface Props {
  login: Offer[];
  authorizationStatus: string;
}

export const App: React.FunctionComponent<Props> = (props: Props) => {
  const {login, authorizationStatus} = props;

  return (
    <BrowserRouter>
      <Switch>
        <Route exact path={AppRoute.ROOT}>
          <Main />
        </Route>
        <Route path={`${AppRoute.PROPERTY}/:id`} render={() => {
          return <Property />;
        }} />
        <Route exact path={AppRoute.AUTH}>
          {(authorizationStatus === AuthorizationStatus.AUTH) ?
            <Redirect to={AppRoute.ROOT} /> :
            <SignIn
              onSubmit={login}
            />
          }
        </Route>
        <PrivateRoute
          exact
          path={AppRoute.FAVORITES}
          render={() => {
            return (<Favorites />);
          }}
        />
      </Switch>
    </BrowserRouter>
  );
};

const mapStateToProps = (state) => ({
  authorizationStatus: getAuthorizationStatus(state),
});

const mapDispatchToProps = (dispatch) => ({
  login(authData) {
    dispatch(Operation.checkAuth(authData));
  },
});

export default connect(mapStateToProps, mapDispatchToProps)(App);
