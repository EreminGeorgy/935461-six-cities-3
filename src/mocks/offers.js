const shortid = require(`shortid`);

// const MOCKS_TOTAL = 10;
// const IMAGES = [
//   `img/apartment-01.jpg`,
//   `img/apartment-02.jpg`,
//   `img/apartment-01.jpg`,
//   `img/apartment-03.jpg`,
//   `img/apartment-02.jpg`,
//   `img/room.jpg`
// ];
// const TITLES = [
//   `Wood and stone place`,
//   `Beautiful &amp; luxurious apartment at great location`,
//   `Canal View Prinsengracht`,
//   `Nice, cozy, warm big bed apartment`,
// ];
// const TYPES = [
//   `Apartment`,
//   `Private room`,
//   `house`,
//   `hotel`,
// ];
// const HOUSEHOLD_ITEMS = [
//   `Wi-Fi`,
//   `Washing machine`,
//   `Towels`,
//   `Heating`,
//   `Coffee machine`,
//   `Baby seat`,
//   `Kitchen`,
//   `Cabel TV`,
//   `Fridge`,
// ];
// const USER_NAMES = [
//   `Angelina`,
//   `Max`,
// ];
// const USER_AVATAR_SRC = [
//   `img/avatar-angelina.jpg`,
//   `img/avatar-max.jpg`,
// ];
// const DESCRIPTION_TEXT = [
//   `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.`,
//   `An independent House, strategically located between Rembrand Square and National Opera, but where the bustle of the city comes to rest in this alley flowery and colorful.`,
// ];

// const randomBoolean = () => Boolean(Math.round(Math.random()));
// const randomNumber = ({min = 0, max = 1}) => {
//   return min + Math.floor(max * Math.random());
// };

// const getMock = () => ({
//   id: shortid.generate(),
//   city: citiesData[Math.floor(Math.random() * 5)],
//   previewSrc: IMAGES[Math.floor(Math.random() * 6)],
//   imagesSrc: IMAGES,
//   title: TITLES[Math.floor(Math.random() * 4)],
//   description: DESCRIPTION_TEXT,
//   isPremium: randomBoolean(),
//   type: TYPES[Math.floor(Math.random() * 4)],
//   raiting: randomNumber({min: 0, max: 5}),
//   bedrooms: randomNumber({min: 1, max: 3}),
//   guests: randomNumber({min: 1, max: 9}),
//   price: randomNumber({min: 0, max: 999}),
//   householdItems: new Set(HOUSEHOLD_ITEMS),
//   host: {
//     name: USER_NAMES[Math.floor(Math.random() * 2)],
//     isPro: randomBoolean(),
//     avatarUrl: USER_AVATAR_SRC[Math.floor(Math.random() * 2)],
//   },
//   location: [
//     [52.370955, 4.873096],
//     [52.379553, 4.873096],
//     [52.390955, 4.909309],
//     [52.360955, 4.909309]
//   ]
// });

// const getOffers = () => {
//   return new Array(MOCKS_TOTAL).fill(``).map(getMock);
// }

// const offers = getOffers();

const offers = [
  {
    id: shortid.generate(),
    city: {
      name: `Cologne`,
      location: [50.939485, 6.948775],
    },
    imagesSrc: [
      `img/apartment-01.jpg`,
      `img/apartment-02.jpg`,
      `img/apartment-01.jpg`,
      `img/apartment-03.jpg`,
      `img/apartment-02.jpg`,
      `img/room.jpg`
    ],
    previewSrc: `img/apartment-01.jpg`,
    title: `_`,
    price: 0,
    raiting: 0,
    type: `Apartment`,
    isPremium: true,
    bedrooms: 3,
    guests: 2,
    householdItems: new Set([
      `Wi-Fi`,
      `Washing machine`,
      `Towels`,
    ]),
    host: {
      name: `Angelina`,
      isPro: true,
      avatarUrl: `img/avatar-angelina.jpg`,
    },
    location: [50.937764, 6.961390],
  },
  {
    id: shortid.generate(),
    city: {
      name: `Paris`,
      location: [48.859335, 2.350730],
    },
    imagesSrc: [
      `img/apartment-01.jpg`,
      `img/apartment-02.jpg`,
      `img/apartment-01.jpg`,
      `img/apartment-03.jpg`,
      `img/apartment-02.jpg`,
      `img/room.jpg`
    ],
    previewSrc: `img/apartment-02.jpg`,
    title: `Wood and stone place`,
    price: 90,
    raiting: 8.35,
    type: `Private room`,
    isPremium: false,
    bedrooms: 3,
    guests: 2,
    householdItems: new Set([
      `Wi-Fi`,
      `Washing machine`,
      `Towels`,
    ]),
    host: {
      name: `Angelina`,
      isPro: true,
      avatarUrl: `img/avatar-angelina.jpg`,
    },
    location: [48.872344, 2.360418],
  },
  {
    id: shortid.generate(),
    city: {
      name: `Brussels`,
      location: [50.855889, 4.351608],
    },
    imagesSrc: [
      `img/apartment-01.jpg`,
      `img/apartment-02.jpg`,
      `img/apartment-01.jpg`,
      `img/apartment-03.jpg`,
      `img/apartment-02.jpg`,
      `img/room.jpg`
    ],
    previewSrc: `img/apartment-03.jpg`,
    title: `Canal View Prinsengracht`,
    price: 80,
    raiting: 8.2,
    type: `1`,
    isPremium: true,
    bedrooms: 3,
    guests: 2,
    householdItems: new Set([
      `Wi-Fi`,
      `Washing machine`,
      `Towels`,
    ]),
    host: {
      name: `Angelina`,
      isPro: true,
      avatarUrl: `img/avatar-angelina.jpg`,
    },
    location: [50.855889, 4.351608]
  },
  {
    id: shortid.generate(),
    city: {
      name: `Amsterdam`,
      location: [52.371512, 4.893750],
    },
    imagesSrc: [
      `img/apartment-01.jpg`,
      `img/apartment-02.jpg`,
      `img/apartment-01.jpg`,
      `img/apartment-03.jpg`,
      `img/apartment-02.jpg`,
      `img/room.jpg`
    ],
    previewSrc: `img/room.jpg`,
    title: `Nice, cozy, warm big bed apartment`,
    price: 50,
    raiting: 3,
    type: `Apartment`,
    isPremium: false,
    bedrooms: 3,
    guests: 2,
    householdItems: new Set([
      `Wi-Fi`,
      `Washing machine`,
      `Towels`,
    ]),
    host: {
      name: `Angelina`,
      isPro: true,
      avatarUrl: `img/avatar-angelina.jpg`,
    },
    location: [52.370955, 4.873096]
  },
  {
    id: shortid.generate(),
    city: {
      name: `Amsterdam`,
      location: [52.371512, 4.893750],
    },
    imagesSrc: [
      `img/apartment-01.jpg`,
      `img/apartment-02.jpg`,
      `img/apartment-01.jpg`,
      `img/apartment-03.jpg`,
      `img/apartment-02.jpg`,
      `img/room.jpg`
    ],
    previewSrc: `img/room.jpg`,
    title: `Nice, cozy, warm big bed apartment`,
    price: 50,
    raiting: 3,
    type: `Apartment`,
    isPremium: false,
    bedrooms: 3,
    guests: 2,
    householdItems: new Set([
      `Wi-Fi`,
      `Washing machine`,
      `Towels`,
    ]),
    host: {
      name: `Angelina`,
      isPro: true,
      avatarUrl: `img/avatar-angelina.jpg`,
    },
    location: [52.379553, 4.873096]
  },
];

export {offers};
