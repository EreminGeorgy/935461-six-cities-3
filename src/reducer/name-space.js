const NameSpace = {
  DATA: `DATA`,
  APPLICATION: `APPLICATION`,
  USER: `USER`,
  COMMENTS: `COMMENTS`,
  FAVORITES: `FAVORITES`,
};

export default NameSpace;
