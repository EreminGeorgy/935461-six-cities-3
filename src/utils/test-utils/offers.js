const offers = [
  {
    id: 1,
    city: {
      name: `Paris`,
      location: [48.859335, 2.350730],
    },
    imagesSrc: [
      `img/apartment-01.jpg`,
      `img/apartment-02.jpg`,
      `img/apartment-01.jpg`,
      `img/apartment-03.jpg`,
      `img/apartment-02.jpg`,
      `img/room.jpg`
    ],
    previewSrc: `img/apartment-01.jpg`,
    title: `_`,
    price: 0,
    raiting: 0,
    type: `Apartment`,
    isPremium: true,
    bedrooms: 3,
    guests: 2,
    householdItems: new Set([
      `Wi-Fi`,
      `Washing machine`,
      `Towels`,
    ]),
    host: {
      name: `Angelina`,
      isPro: true,
      avatarUrl: `img/avatar-angelina.jpg`,
    },
    location: [52.3909553943508, 4.85309666406198],
  },
  {
    id: 2,
    city: {
      name: `Paris`,
      location: [48.859335, 2.350730],
    },
    imagesSrc: [
      `img/apartment-01.jpg`,
      `img/apartment-02.jpg`,
      `img/apartment-01.jpg`,
      `img/apartment-03.jpg`,
      `img/apartment-02.jpg`,
      `img/room.jpg`
    ],
    previewSrc: `img/apartment-02.jpg`,
    title: `Wood and stone place`,
    price: 90,
    raiting: 8.35,
    type: `Private room`,
    isPremium: false,
    bedrooms: 3,
    guests: 2,
    householdItems: new Set([
      `Wi-Fi`,
      `Washing machine`,
      `Towels`,
    ]),
    host: {
      name: `Angelina`,
      isPro: true,
      avatarUrl: `img/avatar-angelina.jpg`,
    },
    location: [52.369553943508, 4.85309666406198],
  },
  {
    id: 3,
    city: {
      name: `Paris`,
      location: [48.859335, 2.350730],
    },
    imagesSrc: [
      `img/apartment-01.jpg`,
      `img/apartment-02.jpg`,
      `img/apartment-01.jpg`,
      `img/apartment-03.jpg`,
      `img/apartment-02.jpg`,
      `img/room.jpg`
    ],
    previewSrc: `img/apartment-03.jpg`,
    title: `Canal View Prinsengracht`,
    price: 80,
    raiting: 8.2,
    type: `1`,
    isPremium: true,
    bedrooms: 3,
    guests: 2,
    householdItems: new Set([
      `Wi-Fi`,
      `Washing machine`,
      `Towels`,
    ]),
    host: {
      name: `Angelina`,
      isPro: true,
      avatarUrl: `img/avatar-angelina.jpg`,
    },
    location: [52.3909553943508, 4.929309666406198]
  },
  {
    id: 4,
    city: {
      name: `Paris`,
      location: [48.859335, 2.350730],
    },
    imagesSrc: [
      `img/apartment-01.jpg`,
      `img/apartment-02.jpg`,
      `img/apartment-01.jpg`,
      `img/apartment-03.jpg`,
      `img/apartment-02.jpg`,
      `img/room.jpg`
    ],
    previewSrc: `img/room.jpg`,
    title: `Nice, cozy, warm big bed apartment`,
    price: 50,
    raiting: 3,
    type: `Apartment`,
    isPremium: false,
    bedrooms: 3,
    guests: 2,
    householdItems: new Set([
      `Wi-Fi`,
      `Washing machine`,
      `Towels`,
    ]),
    host: {
      name: `Angelina`,
      isPro: true,
      avatarUrl: `img/avatar-angelina.jpg`,
    },
    location: [52.3809553943508, 4.939309666406198]
  },
];

const offer = {
  id: 5,
  previewSrc: `img/apartment-01.jpg`,
  imagesSrc: [
    `img/apartment-01.jpg`,
    `img/apartment-02.jpg`,
    `img/apartment-01.jpg`,
    `img/apartment-03.jpg`,
    `img/apartment-02.jpg`,
    `img/room.jpg`
  ],
  title: `title`,
  description: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.`,
  price: 0,
  raiting: 10,
  type: `Apartment`,
  isPremium: true,
  bedrooms: 3,
  guests: 2,
  householdItems: new Set([
    `Wi-Fi`,
    `Washing machine`,
    `Towels`,
  ]),
  host: {
    name: `Angelina`,
    isPro: true,
    avatarUrl: `img/avatar-angelina.jpg`,
  }
};

const city = {
  name: `Paris`,
  location: [48.859335, 2.350730],
};

const cities = [{
  name: `Paris`,
  location: [48.859335, 2.350730],
},
{
  name: `Paris`,
  location: [48.859335, 2.350730],
}];

export {offer, offers, city, cities};
