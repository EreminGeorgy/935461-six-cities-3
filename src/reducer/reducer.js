import {offers} from '../mocks/offers';
import {extend, getCities, getOffersByCity} from '../utils/utils.js';

const citiesList = getCities(offers);
const initialOffers = getOffersByCity(offers, citiesList[0]);

const initialState = {
  activeCity: citiesList[0],
  offersInActiveCity: initialOffers,
  cities: citiesList,
};

const ActionCreator = {
  newCity: (newCity) => ({
    type: `NEW_CITY`,
    payload: newCity,
  }),
  getOffers: (newCity) => ({
    type: `GET_OFFERS`,
    payload: newCity,
  }),
};

const reducer = (state = initialState, action) => {
  switch (action.type) {
    case `NEW_CITY`:
      return extend(state, {
        activeCity: action.payload,
      });
    case `GET_OFFERS`:
      return extend(state, {
        offersInActiveCity: getOffersByCity(offers, action.payload),
      });
  }
  return state;
};

export {reducer, ActionCreator};
