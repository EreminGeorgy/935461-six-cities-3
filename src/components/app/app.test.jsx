import React from "react";
import renderer from "react-test-renderer";
import {App} from "./app.jsx";
import {offers, cities} from "../../utils/test-utils/offers.js";
// import configureStore from "redux-mock-store";
import {Provider} from 'react-redux';
import reducer from '../../reducer/reducer.js';
import {createStore} from 'redux';

const store = createStore(
    reducer
);

const activeCity = cities[0];
const offersInActiveCity = offers;
// const mockStore = configureStore([]);
// const store = mockStore({
//   cities,
//   activeCity: cities[0],
//   offersInActiveCity: offers,
// });

it(`<App /> should render App`, () => {
  const tree = renderer
    .create(
        <Provider store={store}>
          <App
            offersInActiveCity={offersInActiveCity}
            city={activeCity}
          />
        </Provider>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
