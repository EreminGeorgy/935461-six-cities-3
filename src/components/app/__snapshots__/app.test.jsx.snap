// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> should render App 1`] = `
<p>
  No data loaded
</p>
`;
